File: src/main/java/com/jetbrains/gymusserverjava/memberships/dtos/responses/MemberCardResponseDto.java
package com.jetbrains.gymusserverjava.memberships.dtos.responses;

import java.time.LocalDate;

public record MemberCardResponseDto(
        int id,
        String firstName,
        String lastName,
        LocalDate birthdate,
        LocalDate joinDate,
        LocalDate endDate,
        String personalPhoto
) {

    public boolean isActive() {
        return endDate.isAfter(LocalDate.now());
    }

}

File: src/main/java/com/jetbrains/gymusserverjava/memberships/dtos/responses/MemberCardResponseDto.kt
package com.jetbrains.gymusserverjava.memberships.dtos.responses

import java.time.LocalDate

data class MemberCardResponseDto(
    val id: Int,
    val firstName: String,
    val lastName: String,
    val birthdate: LocalDate,
    val joinDate: LocalDate,
    val startDate: LocalDate,
    val endDate: LocalDate,
    val personalPhoto: String
) {
    val isActive: Boolean
        get() = endDate.isAfter(LocalDate.now())
}